enzyme/test/Enzyme/ReverseMode/blas/cblas_dgemm_col_mod1_split.ll

```diff
@@ -0,0 +1,142 @@
+;RUN: %opt < %s %loadEnzyme -enzyme -mem2reg -instsimplify -simplifycfg -S | FileCheck %s
+
+;#include <cblas.h>
+;
+;extern double __enzyme_autodiff(void *, double *, double *, double *, double *, double *, double*, double, double);
+;
+;void outer(double *Afoo, double *Bfoo, double *Cfoo, double alpha, double beta) {
+;    cblas_dgemm(CblasColMajor, CblasNoTrans, CblasNoTrans, 4, 3, 2, alpha, Afoo, 4, Bfoo, 2, beta, Cfoo, 4);
+;}
+;
+;void g(double *A, double *B, double *C, double alpha, double beta) {
+;    outer(A, B, C, alpha, beta);
+;    B[0] = 10;
+;}
+;
+;int main() {
+;    double A[] = {0.11, 0.12, 0.13, 0.14,
+;                 0.21, 0.22, 0.23, 0.24};
+;    double B[] = {1011, 1012,
+;                 1021, 1022,
+;                 1031, 1032};
+;    double C[] = {0.00, 0.00, 0.00, 0.00,
+;                 0.00, 0.00, 0.00, 0.00,
+;                 0.00, 0.00, 0.00, 0.00};
+;    double A1[] = {0, 0, 0, 0, 0, 0, 0, 0};
+;    double B1[] = {0, 0, 0, 0, 0, 0};
+;    double C1[] = {1, 3, 7, 11,
+;                  0.00, 0.00, 0.00, 0.00,
+;                  0.00, 0.00, 0.00, 0.00};
+;    __enzyme_autodiff((void*)g, A, A1, B, B1, C, C1, 2.0, 3.0);
+;}
+
+target datalayout = "e-m:e-i64:64-f80:128-n8:16:32:64-S128"
+target triple = "x86_64-unknown-linux-gnu"
+
+@__const.main.A = private unnamed_addr constant [8 x double] [double 1.100000e-01, double 1.200000e-01, double 1.300000e-01, double 1.400000e-01, double 2.100000e-01, double 2.200000e-01, double 2.300000e-01, double 2.400000e-01], align 16
+
+define dso_local void @outer(double* %Afoo, double* %Bfoo, double* %Cfoo, double %alpha, double %beta) {
+entry:
+  call void @cblas_dgemm(i32 102, i32 111, i32 111, i32 4, i32 3, i32 2, double %alpha, double* %Afoo, i32 4, double* %Bfoo, i32 2, double %beta, double* %Cfoo, i32 4)
+  ret void
+}
+
+declare dso_local void @cblas_dgemm(i32, i32, i32, i32, i32, i32, double, double*, i32, double*, i32, double, double*, i32)
+
+define dso_local void @g(double* %A, double* %B, double* %C, double %alpha, double %beta) {
+entry:
+  call void @outer(double* %A, double* %B, double* %C, double %alpha, double %beta)
+  %arrayidx = getelementptr inbounds double, double* %B, i64 0
+  store double 1.000000e+01, double* %arrayidx, align 8
+  ret void
+}
+
+define dso_local i32 @main() {
+entry:
+  %A = alloca [8 x double], align 16
+  %B = alloca [6 x double], align 16
+  %C = alloca [12 x double], align 16
```

Comment:
Can you remove the main except for the call to autodiff (and perhaps just have arguments forwarded to the autodiff call)